tests/netutils/mock_dns.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief       Mock implementation of sock_dns
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+#include <string.h>
+
+#include "net/af.h"
+#include "net/ipv6/addr.h"
+#include "net/sock/dns.h"
+
+int sock_dns_query(const char *domain_name, void *addr_out, int family)
+{
+    const ipv6_addr_t a = { {
```

Comment:
```suggestion
    static const ipv6_addr_t a = { {
```

might lead to a better stack usage, but I guess this is just premature optimization.

Reply:
`const` should always go to .text / ROM, not on the stack.

Reply:
I think to really enforce it, you need to declare it `static`. But not 100% sure. I guess you can just try and see if the ROM grows when adding `static` :-).